numpy/distutils/ccompiler_opt.py

```diff
@@ -2551,6 +2551,8 @@ def _generate_config(self, output_dir, dispatch_src, targets, has_baseline=False
         except OSError:
             pass
 
+        os.makedirs(os.path.dirname(config_path), exist_ok=True)
+
```

Comment:
Any idea why this is needed now?

Reply:
yeah, it happens when there's no `.c.src` remaining in a setup.py subdirectory, but the system somehow needs that subdirectory to be created.